src/OpenRiaServices.Client/Framework/LoadOperation.cs

```diff
@@ -177,11 +178,44 @@ public sealed class LoadOperation<TEntity> : LoadOperation
         internal LoadOperation(EntityQuery<TEntity> query, LoadBehavior loadBehavior,
             Action<LoadOperation<TEntity>> completeAction, object userState,
             bool supportCancellation)
-            : base(query, loadBehavior, userState, supportCancellation)
+            : base(query, loadBehavior, userState, supportCancellation ? new CancellationTokenSource() : null)
         {
             this._completeAction = completeAction;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LoadOperation"/> class.
+        /// </summary>
+        /// <param name="loadTask"></param>
+        /// <param name="cancellationTokenSource"><see cref="CancellationTokenSource"/> which will be used to request cancellation if <see cref="OperationBase.Cancel()"/> is called, if <c>null</c> then cancellation will not be possible</param>
+        /// <param name="query">The query to load, will set <see cref="LoadOperation.EntityQuery"/>.</param>
+        /// <param name="loadBehavior"><see cref="LoadBehavior"/> used for the load operation, will set <see cref="LoadOperation.LoadBehavior"/>.</param>
+        /// <param name="completeAction">Action to execute when the operation completes.</param>
+        /// <param name="userState">Optional user state for the operation.</param>
+        // TODO: Determine which constructor to make public, if it is the Old above then the Complete method must be made public as well
```

Comment:
I beleive it is better t use the new constructor in this case.

Reply:
I agree, the complete method will be internal only for tests